internal/services/disbursement_management_service.go

```diff
@@ -149,41 +149,79 @@ func (s *DisbursementManagementService) StartDisbursement(ctx context.Context, d
-			return nil
+		if len(receiverWallets) != 0 {
+			eventData := make([]schemas.EventReceiverWalletSMSInvitationData, 0, len(receiverWallets))
+			for _, receiverWallet := range receiverWallets {
+				eventData = append(eventData, schemas.EventReceiverWalletSMSInvitationData{ReceiverWalletID: receiverWallet.ID})
+			}
+
+			sendInviteMsg, msgErr := events.NewMessage(ctx, events.ReceiverWalletNewInvitationTopic, disbursement.ID, events.BatchReceiverWalletSMSInvitationType, eventData)
+			if msgErr != nil {
+				return fmt.Errorf("creating new message: %w", msgErr)
+			}
+
+			msgs = append(msgs, *sendInviteMsg)
+		} else {
+			log.Ctx(ctx).Infof("no receiver wallets to send invitation for disbursement ID %s", disbursementID)
 		}
 
-		msg, err := events.NewMessage(ctx, events.PaymentReadyToPayTopic, disbursementID, events.PaymentReadyToPayDisbursementStarted, nil)
+		payments, err := s.models.Payment.GetReadyByDisbursementID(ctx, dbTx, disbursementID)
 		if err != nil {
-			return fmt.Errorf("creating new message: %w", err)
+			return fmt.Errorf("getting ready payments for disbursement with id %s: %w", disbursementID, err)
 		}
 
-		paymentsReadyToPay := schemas.EventPaymentsReadyToPayData{TenantID: msg.TenantID}
-		for _, payment := range payments {
-			paymentsReadyToPay.Payments = append(paymentsReadyToPay.Payments, schemas.PaymentReadyToPay{ID: payment.ID})
-		}
-		msg.Data = paymentsReadyToPay
+		if len(payments) != 0 {
+			paymentsReadyToPayMsg, msgErr := events.NewMessage(ctx, events.PaymentReadyToPayTopic, disbursementID, events.PaymentReadyToPayDisbursementStarted, nil)
+			if msgErr != nil {
+				return fmt.Errorf("creating new message: %w", msgErr)
+			}
 
-		if s.eventProducer != nil {
-			err := s.eventProducer.WriteMessages(ctx, *msg)
-			if err != nil {
-				return fmt.Errorf("writing message %s on event producer: %w", msg, err)
+			paymentsReadyToPay := schemas.EventPaymentsReadyToPayData{TenantID: paymentsReadyToPayMsg.TenantID}
+			for _, payment := range payments {
+				paymentsReadyToPay.Payments = append(paymentsReadyToPay.Payments, schemas.PaymentReadyToPay{ID: payment.ID})
 			}
+			paymentsReadyToPayMsg.Data = paymentsReadyToPay
+
+			msgs = append(msgs, *paymentsReadyToPayMsg)
 		} else {
-			log.Ctx(ctx).Errorf("event producer is nil, could not publish message %s", msg.String())
+			log.Ctx(ctx).Infof("no payments ready to pay for disbursement ID %s", disbursementID)
+		}
+
+		// If there's no message to publish we only log it.
+		if len(msgs) == 0 {
+			log.Ctx(ctx).Infof("no messages to be published for disbursement ID %s", disbursementID)
+			return nil
+		}
+
+		if err = s.produceEvents(ctx, msgs...); err != nil {
+			return err
 		}
 
 		return nil
 	})
 }
 
+func (s *DisbursementManagementService) produceEvents(ctx context.Context, msgs ...events.Message) error {
+	if s.eventProducer == nil {
+		log.Ctx(ctx).Errorf("event producer is nil, could not publish messages %s", msgs)
```

Comment:
This small change should provide more info when debugging the logs:

```suggestion
		log.Ctx(ctx).Errorf("event producer is nil, could not publish messages %+v", msgs)
```